### PR TITLE
#14 Fix notecard serial error

### LINKED ISSUE #14: Sleep at end of main thread (or option to)
Hello,

When running the exporter, I noticed on a ZeroW it's pretty much pegging the CPU at 50% all the time. Looking at your code I can see that you're trying to gather the sensor data in a really tight loop. For my needs this isn't necessary, so I added a 10s sleep which cuts the CPU usage to essentially zero (~2% when gathering). Prometheus is only scraping every 10s anyway, so updating the sensor data more often isn't necessary.

It would be nice if this was an option, so that we can reduce CPU usage when we don't need the most up-to-date sensor data on every metrics poll.

Alternatively adding a 1s sleep in may be enough to limit CPU usage while still providing up-to-date data - it's unlikely anyone (sensible) is running Prometheus with a sub-second scrape interval.

### PR DESCRIPTION
**Pull request by sighmon**
Resolves #14

- [x] Handle `SerialError` when trying to post to Notecard

### CODE REVIEW

File: enviroplus_exporter.py [hunk 1]
```diff
@@ -78,28 +90,60 @@
...
-
-    if factor:
-        cpu_temps = [get_cpu_temperature()] * 5
-        cpu_temp = get_cpu_temperature()
-        # Smooth out with some averaging to decrease jitter
-        cpu_temps = cpu_temps[1:] + [cpu_temp]
-        avg_cpu_temp = sum(cpu_temps) / float(len(cpu_temps))
-        temperature = raw_temp - ((avg_cpu_temp - raw_temp) / factor)
-    else:
-        temperature = raw_temp
```
Comment (tijmenvandenbrink) [inline enviroplus_exporter.py:102]: @sighmon what's your idea about changing this to a more basic approach?

**Comment by tijmenvandenbrink** [comment]
@sighmon seems to have some conflicts before I can merge them. Would you care to fix them and I will merge the changes. Thanks for your contribution!

**Comment by sighmon** [comment]
@tijmenvandenbrink Sorry, I didn't mean to open this in your repo - I'm still playing around with a few things here, so I'll create a fresh PR when it's ready.

File: enviroplus_exporter.py (hunk 1 above)
Comment (sighmon) [reply enviroplus_exporter.py:102]: @tijmenvandenbrink It wasn't working out linearly across the temperature range for me and also didn't take into account the humidity changes from the extra heat. My simplistic version isn't great either, so I'll keep working on it.

**Pull request closed without merging**